### PR TITLE
Move some tests to quarantine

### PR DESCRIPTION
**Pull request by potiuk**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: tests/jobs/test_backfill_job.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
 DEFAULT_DATE = timezone.datetime(2016, 1, 1)
 
 
+@pytest.mark.quarantined
 class TestBackfillJob(unittest.TestCase):
```
Comment (turbaszek) [inline tests/jobs/test_backfill_job.py:56]: I dunno... it seems to impose a lot of risk to skip **this** class

Comment (potiuk) [reply tests/jobs/test_backfill_job.py:56]: Sure. I will create an issue assign you to it first :)

Comment (potiuk) [reply tests/jobs/test_backfill_job.py:56]: But on a more serious note - > our goal is to clean up everything from Quarantine before the release. I think for master we can afford some temporary lack of testing.

Side note - the way we have the tests now - they are skipping all Py3.7 tests :(. And we are merging failing tests anyway. I think we should simply - once we get stable have a project where we have all classes in Quarantine, assign people and make them move out of quarantine.

Comment (turbaszek) [reply tests/jobs/test_backfill_job.py:56]: I know. I just wanted to express my concerns. Let's move on and fix those tests asap

**Review by turbaszek — APPROVED** [review]

**Pull request merged**